File: CS50x/problem_sets-2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
// to count index and print text's grade
void calculator(int l, int w, int t);

int main(void)
{
    string text = get_string("Text: ");
    int l = count_letters(text);
    int w = count_words(text);
    int t = count_sentences(text);
    calculator(l, w, t);
}

int count_letters(string s)
{
    int letters = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        char x = s[i];
        if ((x >= 'a' && x <= 'z') || (x >= 'A' && x <= 'Z'))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string s)
{
    int words = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        char x = s[i];
        if (x == ' ')
        {
            words++;
        }
    }
    words++;
    return words;
}

int count_sentences(string s)
{
    int sentences = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        char x = s[i];
        if (x == '!' || x == '?' || x == '.')
        {
            sentences++;
        }
    }
    return sentences;
}

void calculator(int l, int w, int t)
{
    double L = ((float) l / w) * 100;
    double S = ((float) t / w) * 100;
    // "Coleman-Liau index" formula
    double index = 0.0588 * L - 0.296 * S - 15.8;
    // round up the index
    double grade = round(index);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) grade);
    }
}

File: CS50x/problem_sets-6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("Text: ")


def main():
    a = count_letters(text)
    b = count_words(text)
    c = count_sentences(text)
    # to count index and print text's grade
    calculator(a, b, c)


def count_letters(txt):
    letter_counter = 0
    for i in txt:
        if (i >= "a" and i <= "z") or (i >= "A" and i <= "Z"):
            letter_counter += 1

    return letter_counter


def count_words(txt):
    word_counter = 1
    for i in txt:
        if i == " ":
            word_counter += 1

    return word_counter


def count_sentences(txt):
    sntc_counter = 0
    for i in txt:
        if i == "?" or i == "!" or i == ".":
            sntc_counter += 1

    return sntc_counter


def calculator(lttr, wrd, snt):
    L = 100 * (lttr / wrd)
    S = 100 * (snt / wrd)

    # "Coleman-Liau index" formula
    index = 0.0588 * L - 0.296 * S - 15.8

    # round up the index
    grade = round(index)

    if grade >= 16:
        print("Grade 16+")
    elif grade <= 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


main()